app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -1,32 +1,25 @@
 module EmsRefresh
   module Refreshers
     module EmsRefresherMixin
+      def format_ems_for_logging(ems)
+        "EMS: [#{ems.name}], id: [#{ems.id}]"
+      end
+
       def refresh
-        _log.info "Refreshing all targets..."
+        preprocess_targets
 
         @targets_by_ems_id.each do |ems_id, targets|
           # Get the ems object
           ems = @ems_by_ems_id[ems_id]
-
-          log_ems_target = "EMS: [#{ems.name}], id: [#{ems.id}]"
-          _log.info "#{log_ems_target} Refreshing targets for EMS: [#{ems.name}], id: [#{ems.id}]..."
-          targets.each { |t| _log.info "#{log_ems_target}   #{t.class} [#{t.name}] id [#{t.id}]" }
-
           ems_refresh_start_time = Time.now
 
           begin
-            _log.debug "#{log_ems_target} Parsing #{refresher_type} inventory..."
-            hashes = parse_inventory(ems, targets)
-            _log.debug "#{log_ems_target} Parsing #{refresher_type} inventory..." \
-                       "Completed in #{Time.now - ems_refresh_start_time} seconds"
-            _log.debug "#{log_ems_target} inv hashes:\n#{hashes.pretty_inspect}" if self.class::DEBUG_TRACE
-
-            if hashes.blank?
-              # TODO: determine if this is "success" or "failed"
-              _log.warn "No inventory data returned for EMS: [#{ems.name}], id: [#{ems.id}]..."
-              next
-            end
-            save_inventory(ems, targets, hashes)
+            _log.info "Refreshing all targets..."
+            log_ems_target = format_ems_for_logging(ems)
+            _log.info "#{log_ems_target} Refreshing targets for EMS..."
+            targets.each { |t| _log.info "#{log_ems_target}   #{t.class} [#{t.name}] id [#{t.id}]" }
```

Comment:
It's currently printed in production (at least for VMWare).  But, yeah this could be optimized.

I'd rather save that for a different PR because this is maintaining current state and I don't want to disrupt refresh logging *too* much in this one PR.